x-pack/legacy/plugins/siem/server/lib/framework/kibana_framework_adapter.ts

```diff
@@ -21,125 +26,113 @@ import {
+      ...params,
+      ignore_throttled: !includeFrozen,
+      ...(maxConcurrentShardRequests > 0
+        ? { max_concurrent_shard_requests: maxConcurrentShardRequests }
+        : {}),
     });
   }
 
   public registerGraphQLEndpoint(routePath: string, schema: GraphQLSchema): void {
-    this.server.route({
-      options: {
-        tags: ['access:siem'],
+    this.router.get(
+      {
+        path: routePath,
+        validate: { query: configSchema.object({}, { allowUnknowns: true }) },
+        options: {
+          tags: ['access:siem'],
+        },
       },
-      handler: async (request: RequestFacade, h: ResponseToolkit) => {
+      async (context, request, response) => {
         try {
-          const query =
-            request.method === 'post'
-              ? (request.payload as Record<string, any>) // eslint-disable-line @typescript-eslint/no-explicit-any
-              : (request.query as Record<string, any>); // eslint-disable-line @typescript-eslint/no-explicit-any
-
+          const { query } = request;
           const gqlResponse = await runHttpQuery([request], {
-            method: request.method.toUpperCase(),
+            method: 'GET',
             options: (req: RequestFacade) => ({
-              context: { req: wrapRequest(req) },
+              context: { req: wrapRequest(req, context) },
               schema,
             }),
-
             query,
           });
 
-          return h.response(gqlResponse).type('application/json');
+          return response.ok({
+            body: gqlResponse,
+            headers: {
+              'content-type': 'application/json',
+            },
+          });
         } catch (error) {
-          if ('HttpQueryError' !== error.name) {
-            const queryError = Boom.boomify(error);
-
-            queryError.output.payload.message = error.message;
-
-            return queryError;
-          }
-
-          if (error.isGraphQLError === true) {
-            return h
-              .response(error.message)
-              .code(error.statusCode)
-              .type('application/json');
-          }
-
-          const genericError = new Boom(error.message, { statusCode: error.statusCode });
-
-          if (error.headers) {
-            Object.keys(error.headers).forEach(header => {
-              genericError.output.headers[header] = error.headers[header];
-            });
-          }
-
-          // Boom hides the error when status code is 500
-          genericError.output.payload.message = error.message;
-
-          throw genericError;
+          return this.handleError(error, response);
         }
-      },
-      method: ['GET', 'POST'],
-      path: routePath,
-      vhost: undefined,
-    });
+      }
+    );
 
-    if (!this.envMode.prod) {
-      this.server.route({
+    this.router.post(
+      {
+        path: routePath,
+        validate: { body: configSchema.object({}, { allowUnknowns: true }) },
```

Comment:
Because of the many different request formats that come through this route, I'm inclined to punt on typing them for now. Thoughts?

Reply:
not doing validation doesn't make it worse than before, but validation is our most important defence against prototype pollution attacks, so I think you should at least create an issue to track the work to implement the validation to be sure it doesn't get forgotten.

Reply:
Added some initial validation in 22b2535aefab2c05349d3daf1057e920e40b73d2.

I'm gonna see if @XavierM can't come to a reasonable solution here tomorrow, and I'll create an issue based on the results of that.

Reply:
Given that graphQL is slowly being phased out, we're opting for some superficial validation for now while we port things over to rest routes.